packages/vc-verification/src/revocation-verification.ts

```diff
@@ -26,77 +44,101 @@ export class RevocationVerification {
+    } catch (e) {
+      if (!(e instanceof CredentialRevoked)) {
+        throw new Error((e as Error).message);
+      }
     }
-    return await this.verifyRevokerAuthority(namespace, revokerDID);
+
+    const revokerDID = issuerDID(revoker);
+    await this.verifyRevoker(revokerDID, role);
   }
 
   /**
-   * Verifies revoker's authority
-   * @param namespace for which the revoker's authority is to be verified
-   * @param revokerDID whose authority needs to be verified
-   * @returns
+   * Verifies that revoker is authorized to revoke role credential
+   * @param role fully qualified role name
+   * @param revoker DID of revoker
    */
-  async verifyRevokerAuthority(
-    namespace: string,
-    revokerDID: string
-  ): Promise<boolean> {
-    const revokers = await this._revokerResolver.getRevokerDefinition(
-      namespace
-    );
-    if (revokers?.did) {
-      const revokerAddr = addressOf(revokerDID);
-      for (let i = 0; i < revokers.did.length; i++) {
-        const roleRevokerAddr = addressOf(revokers.did[i]);
-        if (roleRevokerAddr.toUpperCase() === revokerAddr.toUpperCase()) {
-          return true;
-        }
-      }
+  private async verifyRevoker(revoker: string, role: string) {
+    const revokers = await this._revokerResolver.getRevokerDefinition(role);
+    if (!revokers) {
+      throw new NoRevokers(role);
     }
-    let isCredentialVerified = false;
-    let isCredentialIssuerVerified = false;
-    if (revokers?.roleName) {
-      const vc = await this._credentialResolver.getCredential(
-        revokerDID,
-        revokers.roleName
-      );
-      if (vc) {
-        isCredentialVerified = await verifyCredential(
-          JSON.stringify(vc),
-          JSON.stringify({})
-        );
-        isCredentialIssuerVerified =
-          await this._issuerVerification.verifyIssuerAuthority(
-            revokers.roleName,
-            vc.issuer as string
-          );
-      } else {
-        throw new Error('No authoritative credential found for revoker');
+    if (revokers.did) {
+      if (
+        !revokers.did.some((r) => r.toUpperCase() === revoker.toUpperCase())
+      ) {
+        throw new RevokerNotAuthorized(revoker, role);
+      }
+    } else if (revokers.roleName) {
+      try {
+        await this.verifyEnrolment(revoker, revokers.roleName);
+      } catch (e) {
+        throw new RevokerNotAuthorized(revoker, role, (<Error>e).message);
       }
+    } else {
+      throw new InvalidRevokerType(role, revokers?.revokerType);
     }
-    return (
-      isCredentialIssuerVerified && typeof isCredentialVerified === 'string'
-    );
   }
 
   /**
-   * Checks if the purpose of the issued credential is revocation
-   * @param purpose statusPurpose from StatusListCredential
-   * @returns
+   * Verifies that role credential was issued to subject by authorized issuer
+   * @param subject DID of revoker
+   * @param role fully qualified role name
+   *
+   * @todo reuse in VCIssuerVerification
```

Comment:
Maybe we can move this to `VCIssuerVerification` and reuse it here. This way we won't need to initialise `VCIssuerVerification` at all [here](https://github.com/energywebfoundation/ew-credentials/blob/a1b7cb3e16a8416f3bf7c046e3151d2bfec41e26/packages/vc-verification/src/revocation-verification.ts#L40)